### ISSUE TITLE
Omniscient 4.1.0 breaks getDefaultProps

### ISSUE THREAD

**Issue by ChristiaanScheermeijer** [omniscientjs/omniscient#126]
After the 4.1.0 update React components don't receive the default props anymore from the `getDefaultProps()` lifecycle method. After some debugging we've found out that there goes something wrong in the `.type` assignment of the component.

```
create.type = Component;
```

This results in a Component with 2 types nested `component.type.type` where the defaultProps are defined. `React.createElement` only looks in the first `type` for the defaultProps object.

**Comment by torgeir** [comment]
Damn. Sorry about that!

It looked to good to be true, so it usually is. We'll look into this asap. Do you mind providing an example to reproduce the issue you are seeing? We'll convert it to a test and make it pass.

@mikaelbr maybe we ought to look into `create.prototype = Component.prototype` instead as mentioned in https://github.com/omniscientjs/omniscient/pull/125

**Comment by ChristiaanScheermeijer** [comment]
No problem :-)

```
var lifecycleMethods = {

  getDefaultProps: function () {

    return {
      direction: 'vertical'
    };
  }
};

module.exports = component(lifecycleMethods, function ({ children, direction, ...rest }) {
  
  // direction is only defined when given as prop
  console.log(direction);
  
  // same for this.props
  console.log(this.props);

  return <div { ...rest }>{ children }</div>
});
```

**Comment by torgeir** [comment]
This is still an issue after removing `create.type`, the jsx assert fails https://github.com/omniscientjs/omniscient/compare/41260c03faffb20bf3d1917c67899afd4554e84f...issue-126#diff-85be81e3538eeba3d115e6ea83f12809R749

**Comment by torgeir** [comment]
> This results in a Component with 2 types nested component.type.type where the defaultProps are defined

Yeah, this is unfortunate, but "intentional" to resolve an issue with omniscient and jsx w/react 0.14. 

I'm suspecting you never got any `getDefaultProps` after omniscient@4? A fix is in master, expect a patch any minute

**Comment by mikaelbr** [comment]
This should be fixed in the patch release `omniscient@4.1.1`! Hope this works better for you.

**Comment by ChristiaanScheermeijer** [comment]
@torgeir Thanks, we used version 4.0.0 which did work.

@mikaelbr Thanks, I've quickly tested 4.1.1 and the props are now set correctly!
